core/include/drivers/nvmem.h

```diff
@@ -0,0 +1,190 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2023, Microchip
+ */
+
+#ifndef __DRIVERS_NVMEM_H
+#define __DRIVERS_NVMEM_H
+
+#include <kernel/dt_driver.h>
+#include <tee_api_types.h>
+#include <types_ext.h>
+
+struct nvmem_cell;
+
+/*
+ * struct nvmem_ops - Nvmem device driver operations
+ * @read_cell: Allocate @data and load @len bytes to from an nvmem cell
```

Comment:
I wonder if it is a good idea to have this function to allocate in the heap.
Either rename it explicitly:
` * @alloc_read_cell: Allocate @data in the heap and load @len bytes to from an nvmem cell`
Or change the function behaviour to let caller allocate (and possible free) by its own means the retrieved cells:
```c
	/*
	 * Return TEE_ERROR_SHORT_BUFFER if @data if NULL or @len to short. In those cases,
	 * output @len provides the number of cells expected from @data reference.
	TEE_Result (*read_cell)(struct nvmem_cell *cell, uint8_t **data,
				size_t *len);